app/lib/features/bluetooth/backend/bluetooth_backend.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+import 'package:blood_pressure_app/features/bluetooth/backend/bluetooth_low_energy/ble_manager.dart';
+import 'package:blood_pressure_app/features/bluetooth/backend/flutter_blue_plus/fbp_manager.dart';
+import 'package:blood_pressure_app/features/bluetooth/backend/flutter_blue_plus/flutter_blue_plus_mockable.dart';
+import 'package:blood_pressure_app/features/bluetooth/backend/mock/mock_manager.dart';
+import 'package:blood_pressure_app/logging.dart';
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+
+part 'bluetooth_connection.dart';
+part 'bluetooth_device.dart';
+part 'bluetooth_discovery.dart';
+part 'bluetooth_manager.dart';
+part 'bluetooth_service.dart';
+part 'bluetooth_state.dart';
+part 'bluetooth_utils.dart';
+
+/// Supported bluetooth backend implementations
+///
+/// Enum values provide a [create] instance method to easily get a new backend instance
+enum BluetoothBackend {
```

Comment:
A factory method would be more explicit and the intuitive place to look for an implementation.

Reply:
You're right, it doesn't work with a factory method, but you can use a static method which has similar usability.
```dart
static BluetoothManager create([bool legacyMode = false]) {
  if (isTestingEnvironment) {
    return MockBluetoothManager();
  } else if (legacyMode) {
    return FlutterBluePlusManager();
  }
  return BluetoothLowEnergyManager();
}
```

Since I don't want to annoy/pressure you during review and both take similar time: Do you want to consider and implement my suggestions yourself or should I commit formatting/pattern things myself?